### PR TITLE
Simplify Metrics FluentD Pipeline

### PR DESCRIPTION
**Pull request by frankreno**
simplify fluentD routing, make it easier to add custom metrics and update steps in readme with details on how to do this with a seperate HTTP source.

### CODE REVIEW

**Comment by frankreno** [comment]
> LGTM, thanks Frank! Minor nit though, can we change the tab size from 4 to 2 for the stuff within the `<label @DATAPOINT>` block?

Good catch! fixed

File: deploy/README.md [hunk 1]
```diff
@@ -362,24 +364,81 @@ spec:
...
 
-#### Step 3: Update the prometheus-overrides.yaml file to forward the metrics to Sumo.
+#### Step 3: Create a new HTTP source in Sumo Logic.
 
-The `prometheus-overrides.yaml` file controls what metrics get forwarded on to Sumo Logic. In order to get your custom metrics sending into Sumo Logic, you need to update the `prometheus-overrides.yaml` file to include a rule to forward on your custom metrics. Here is an example addition to the `prometheus-overrides.yaml` that will forward metrics to Sumo:
+To avoid [blacklisting](https://help.sumologic.com/Metrics/Understand_and_Manage_Metric_Volume/Blacklisted_Metrics_Sources) metrics should be distributed across multiple HTTP sources.  You can [follow these steps](https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source) to create a new HTTP source for your custom metrics. Make note of the URL as you will need it in the next step.
+
+#### Step 4: Update the metrics.conf FluentD Configuration
+
+Next, you will need to update the fluentD configuration to ensure FluentD routes your custom metrics to the HTTP source you created in the previous step.
```
Comment (drduke1) [inline deploy/README.md]: Should we be consistent with how we write `fluentd`?

File: deploy/README.md [hunk 2]
```diff
@@ -362,24 +364,81 @@ spec:
...
 Once you have created this yaml file, go ahead and run `kubectl create -f name_of_yaml.yaml -n sumologic`. This will create the service monitor in the sumologic namespace.
 
-#### Step 3: Update the prometheus-overrides.yaml file to forward the metrics to Sumo.
+#### Step 3: Create a new HTTP source in Sumo Logic.
 
-The `prometheus-overrides.yaml` file controls what metrics get forwarded on to Sumo Logic. In order to get your custom metrics sending into Sumo Logic, you need to update the `prometheus-overrides.yaml` file to include a rule to forward on your custom metrics. Here is an example addition to the `prometheus-overrides.yaml` that will forward metrics to Sumo:
+To avoid [blacklisting](https://help.sumologic.com/Metrics/Understand_and_Manage_Metric_Volume/Blacklisted_Metrics_Sources) metrics should be distributed across multiple HTTP sources.  You can [follow these steps](https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source) to create a new HTTP source for your custom metrics. Make note of the URL as you will need it in the next step.
```
Comment (drduke1) [inline deploy/README.md]: There is an extra space before "You can follow these steps to create..."

File: deploy/README.md [hunk 3]
```diff
@@ -362,24 +364,81 @@ spec:
...
+
+#### Step 4: Update the metrics.conf FluentD Configuration
+
+Next, you will need to update the fluentD configuration to ensure FluentD routes your custom metrics to the HTTP source you created in the previous step.
+
+  * First, base64 encode the HTTP source URL from the previous step by running `echo <HTTP_SOURCE_URL> | base64`.  Replace `<HTTP_SOURCE_URL>` with the URL from step 3.
+  * Next, you can edit the secret that houses all the HTTP sources URLs. Assuming you installed the collection in the  `sumologic` namespace, you can run `kubectl -n sumologic edit secret sumologic` or edit the YAML you deployed when you set up collection..
```
Comment (drduke1) [inline deploy/README.md]: I think you meant: "Assuming you installed the **collector** in ... "

File: deploy/README.md [hunk 4]
```diff
@@ -362,24 +364,81 @@ spec:
...
+  * In the `data` section, add a new key and the base64 encoded value you created. The following is just a snipped of the secret for an example. Do not alter the existing content, you simply want to add a new key.
```
Comment (drduke1) [inline deploy/README.md]: snippet*

File: deploy/README.md [hunk 5]
```diff
@@ -362,24 +364,81 @@ spec:
...
+  
+```yaml
+data:
+...
+my-custom-metrics: <base64EncodedURL>
+kind: Secret
+```
+
+  * Next you need to edit the FluentD Deployment and add a new environment variable, pointing to the new secret.  Assuming you installed the collection in the  `sumologic` namespace, you can run `kubectl -n sumologic edit deployment fluentd` or edit the YAML you deployed when you set up collection. Note, if you installed using helm, the name of the deployment may be different depending on how you installed the helm chart.
```
Comment (drduke1) [inline deploy/README.md]: "installed the **collector** in the ..."

File: deploy/README.md [hunk 6]
```diff
@@ -362,24 +364,81 @@ spec:
...
+        - name: MY_CUSTOM_METRICS
+          valueFrom:
+            secretKeyRef:
+              key: my-custom-metrics
+              name: sumologic
+        - name: LOG_FORMAT
+          value: fields
+```
+
+  * Finally, you need yo modify the fluentD config to route data to your newly created HTTP source. Assuming you installed the collection in the  `sumologic` namespace, you can run `kubectl -n sumologic edit configmap fluentd` or edit the YAML you deployed when you set up collection. Note, if you installed using helm, the name of the deployment may be different depending on how you installed the helm chart.
```
Comment (drduke1) [inline deploy/README.md]: yo* to
collection* collector

File: deploy/README.md [hunk 7]
```diff
@@ -362,24 +364,81 @@ spec:
...
+  * Locate the section `match prometheus.metrics` and you will insert a new section above this.  The `match` statement should end with a tag that identifies your data that fluentD will use for routing.  Then make sure you point to the environment variable you added to your deployment. The following is an example.
```
Comment (drduke1) [inline deploy/README.md]: Extra space before "The 'match' statement..." and "Then make sure you point..."

File: deploy/README.md [hunk 8]
```diff
@@ -362,24 +364,81 @@ spec:
...
+             @type sumologic
+             @id sumologic.endpoint.metrics
+             endpoint "#{ENV['SUMO_ENDPOINT_METRICS']}"
+             @include metrics.output.conf
+           </match>
+```
+
+#### Step 5: Update the prometheus-overrides.yaml file to forward the metrics to FluentD.
+
+The `prometheus-overrides.yaml` file controls what metrics get forwarded on to Sumo Logic. In order to get your custom metrics sending into Sumo Logic, you need to update the `prometheus-overrides.yaml` file to include a rule to forward on your custom metrics. Make sure you include the same tag you created in your FluentD configmap in the previous step. Here is an example addition to the `prometheus-overrides.yaml` that will forward metrics to Sumo:
```
Comment (drduke1) [inline deploy/README.md]: Change "In order to get your custom metrics sending into Sumo Logic, you need to update"
to
"To send custom metrics to Sumo Logic you need to update..."

File: deploy/README.md [hunk 9]
```diff
@@ -27,8 +27,10 @@ This page has instructions for collecting Kubernetes logs, metrics, and events;
             - [Trim and relabel metrics](#trim-and-relabel-metrics)
         - [Custom Metrics](#custom-metrics)
             - [Step 1: Expose a `/metrics` endpoint on your service](#step-1-expose-a-metrics-endpoint-on-your-service)
-            - [Step 2: Setup a service monitor so that Prometheus pulls the data](#step-2-setup-a-service-monitor-so-that-prometheus-pulls-the-data)
-            - [Step 3: Update the prometheus-overrides.yaml file to forward the metrics to Sumo.](#step-3-update-the-prometheus-overridesyaml-file-to-forward-the-metrics-to-sumo)
+            - [Step 2: Set up a service monitor so that Prometheus pulls the data](#step-2-set-up-a-service-monitor-so-that-prometheus-pulls-the-data)
+            - [Step 3: Step 3: Create a new HTTP source in Sumo Logic.](#step-3-create-a-new-http-source-in-sumo-logic)
```
Comment (rvmiller89) [inline deploy/README.md]: `Step 3: ` is written twice here

File: deploy/README.md [hunk 10]
```diff
@@ -362,26 +364,83 @@ spec:
...
+           </match>
+```
+
+#### Step 5: Update the prometheus-overrides.yaml file to forward the metrics to FluentD.
+
+The `prometheus-overrides.yaml` file controls what metrics get forwarded on to Sumo Logic. To send custom metrics to Sumo Logic you need to update the `prometheus-overrides.yaml` file to include a rule to forward on your custom metrics. Make sure you include the same tag you created in your FluentD configmap in the previous step. Here is an example addition to the `prometheus-overrides.yaml` file that will forward metrics to Sumo:
 
 ```
-- url: http://fluentd:9888/prometheus.metrics
+- url: http://fluentd.sumologic.svc.cluster.local:9888/prometheus.metrics.YOUR_TAG
```
Comment (rvmiller89) [inline deploy/README.md]: base URL should be `http://collection-sumologic.sumologic.svc.cluster.local:9888`

File: deploy/README.md (hunk 9 above)
Comment (frankreno) [reply deploy/README.md]: Fixed

File: deploy/README.md (hunk 10 above)
Comment (frankreno) [reply deploy/README.md]: good call, fixed this.

File: deploy/README.md (hunk 9 above)
Comment (rvmiller89) [reply deploy/README.md]: did you commit the change? still written twice

Comment (frankreno) [reply deploy/README.md]: hrmm its weird, i don't see this locally

**Pull request merged**